src/transformStateWithClones.js

```diff
@@ -7,7 +7,31 @@
  * @return {Object[]}
  */
 function transformStateWithClones(state, actions) {
-  // write code here
+  const stateClone = { ...state };
+  const result = [];
+
+  for (const action of actions) {
+    if (action.type === 'addProperties') {
+      Object.assign(stateClone, action.extraData);
+      result.push({ ...stateClone });
+    }
+
+    if (action.type === 'removeProperties') {
+      for (const key of action.keysToRemove) {
+        delete stateClone[key];
+      }
+      result.push({ ...stateClone });
+    }
+
+    if (action.type === 'clear') {
+      for (const key in stateClone) {
+        delete stateClone[key];
+      }
+      result.push({ ...stateClone });
+    }
```

Comment:
Consider using a switch statement instead of multiple if statements when you have a limited set of conditions to check. This will make the code cleaner and easier to read.